invenio_communities/assets/semantic-ui/js/invenio_communities/settings/profile/DeleteButton.js

```diff
@@ -9,43 +9,78 @@
 
 import { i18next } from "@translations/invenio_communities/i18next";
 import React, { useEffect, useState } from "react";
-import { Button, Icon, Modal } from "semantic-ui-react";
+import { Button, Icon, Modal, Message, Grid, Checkbox, Input } from "semantic-ui-react";
 import PropTypes from "prop-types";
+import { Trans } from "react-i18next";
+import { communityErrorSerializer } from "../../api/serializers";
+import { ErrorMessage } from "react-invenio-forms";
 
 export const DeleteButton = (props) => {
   const [modalOpen, setModalOpen] = useState(false);
   const [loading, setLoading] = useState(false);
-  const cancelBtnRef = React.createRef();
+  const [isChecked1, setIsChecked1] = useState(false);
+  const [isChecked2, setIsChecked2] = useState(false);
+  const [isChecked3, setIsChecked3] = useState(false);
+  const [inputValue, setInputValue] = useState("");
+  const [error, setError] = useState(undefined);
+
+  const checkboxRef = React.createRef();
   const openModalBtnRef = React.createRef();
 
+  const handleInputChange = (event) => {
+    setInputValue(event.target.value);
+  };
+  const handleCheck1Change = () => setIsChecked1(!isChecked1);
+  const handleCheck2Change = () => setIsChecked2(!isChecked2);
+  const handleCheck3Change = () => setIsChecked3(!isChecked3);
+  const handleButtonDisabled = (slug) =>
+    !(isChecked1 && isChecked2 && isChecked3 && inputValue === slug);
+
   const handleOpen = () => setModalOpen(true);
 
   const handleClose = () => {
     setModalOpen(false);
     setLoading(false);
     openModalBtnRef?.current?.focus();
+    setIsChecked1(false);
+    setIsChecked2(false);
+    setIsChecked3(false);
+    setInputValue("");
+    setError(undefined);
   };
 
   const handleDelete = async () => {
     setLoading(true);
-    const { onDelete, redirectURL, onError } = props;
+    const { onDelete, redirectURL } = props;
     try {
       await onDelete();
       if (redirectURL) {
         window.location.href = redirectURL;
       }
     } catch (error) {
-      onError(error);
+      const { message } = communityErrorSerializer(error);
+
+      if (message) {
+        setError(message);
+        setLoading(false);
+      }
+      return;
     }
+
     handleClose();
   };
 
   useEffect(() => {
-    if (modalOpen) cancelBtnRef?.current?.focus();
-  }, [modalOpen, cancelBtnRef]);
-
-  const { label, confirmationMessage } = props;
+    if (modalOpen) {
+      const {
+        current: { inputRef },
+      } = checkboxRef;
+      inputRef.current.focus();
+    }
+  }, [modalOpen]);
 
+  const { label, community } = props;
+  const communitySlug = community.slug;
```

Comment:
what do we gain by this new variable?

Reply:
it is used it 3 places and it's problematic to put `{{ community.slug }}` into the `Trans` component